core/pkg/server/handler.go

```diff
@@ -161,7 +161,37 @@ func (h *Handler) Do(inChan <-chan *spb.Record) {
 	h.logger.Info("handler: started", "stream_id", h.settings.RunId)
 	for record := range inChan {
 		h.logger.Debug("handle: got a message", "record_type", record.RecordType, "stream_id", h.settings.RunId)
+
 		h.handleRecord(record)
+
+		switch record.RecordType.(type) {
+		case *spb.Record_Final:
+		case *spb.Record_Footer:
+		case *spb.Record_NoopLinkArtifact:
+		default:
+			// No-op.
+
```

Comment:
this looks a bit weird with `default` here

Reply:
I wanted to group the no-ops together; should I move this to the bottom?

Reply:
yeah, i think it is more familiar to read it this way

Reply:
done